ugv_nav4d_ros2:
  ros__parameters:
    read_cloud_from_ply: false
    read_pose_from_topic: true
    map_ply_path: 'path to ply file'
    #Frame names to get the TF Transform robot2map
    robot_frame: husky/base_link
    world_frame: map
    #Grid resolution used for primitive and trversability parameters
    grid_resolution: 0.3
    grid_max_x: 100
    grid_min_x: -100
    grid_max_y: 100
    grid_min_y: -100
    dumpOnError: 0
    dumpOnSuccess: 0
    initialPatchRadius: 3.0
    #Mobility Parameters
    maxMotionCurveLength: 100.0
    minTurningRadius: 1.0
    multiplierBackward: 3.0
    multiplierBackwardTurn: 4.0
    multiplierForward: 1.0
    multiplierForwardTurn: 2.0
    multiplierLateral: 4.0
    multiplierLateralCurve: 4.0
    multiplierPointTurn: 3.0
    rotationSpeed: 1.0
    searchProgressSteps: 0.1
    searchRadius: 0.0
    translationSpeed: 1.0   
    #Planner Parameters
    epsilonSteps: 2
    initialEpsilon: 64
    numThreads: 8
    planningTime: 50000000
    #Motion Primitive Parameters:
    cellSkipFactor: 3
    destinationCircleRadius: 10
    generateBackwardMotions: true
    generateForwardMotions: true
    generateLateralMotions: false
    generatePointTurnMotions: false
    numAngles: 42
    numEndAngles: 21
    splineOrder: 4
    #Traversability Parameters:
    allowForwardDownhill: true
    costFunctionDist: 0.0
    distToGround: 0.0
    enableInclineLimitting: false
    inclineLimittingLimit: 0.1
    inclineLimittingMinSlope: 0.2
    initialPatchVariance: 0.0001
    maxSlope: 0.45
    maxStepHeight: 0.2
    minTraversablePercentage: 0.4
    robotHeight: 1.0
    robotSizeX: 0.9
    robotSizeY: 0.6
    slopeMetric: :NONE
    slopeMetricScale: 1.0